client/Containers/AllProducts.jsx

```diff
@@ -1,28 +1,117 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
-import fetchProductList from '../store/product/productList';
+import { fetchProductList } from '../store/product/productList';
+import { setProductPagination } from '../store/product/productPagination';
+import { setProductFilters } from '../store/product/productFilters';
+import PaginationControl from './PaginationControl';
+import FilterSortControl from './FilterSortControl';
+import getProductQueries from '../components/utils/getProductQueries';
+import buildProductQuery from '../components/utils/buildProductQuery';
+
+const mapStateToProps = (
+  { productList, productPagination, productFilters },
+  { location }
+) => ({
+  productList,
+  productPagination,
+  productFilters,
+  location,
+});
+const mapDispatchToProps = (dispatch) => ({
+  getProducts: (query) => dispatch(fetchProductList(query)),
+  setPagination: (productPagination) =>
+    dispatch(setProductPagination(productPagination)),
+  setFilters: (productFilters) => dispatch(setProductFilters(productFilters)),
+});
```

Comment:
I find that it's easier to parse how a component gets and uses things from the store when you place the `mapStateToProps` and `mapDispatchToProps` at the top of the file.